### ISSUE TITLE
new rule - useless if statement/expression

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#9767]
ruff should be able to detect if the code on both sides of an if statement are identical, and report it as unnecessary 
```py
if 1: # error: both sides of the if statement are the same
    print(1)
else:
    print(1)

asdf = 1 if bool() else 2 # error: both sides of the if expression are the same
```

**Comment by sanxiyn** [comment]
This is similar to `if-with-same-arms` (SIM114) in `flake8-simplify`, so I filed a rule request upstream.

**Comment by mikeleppane** [comment]
WIP

**Comment by DetachHead** [comment]
[`useless-if-else` (`RUF034`)](https://docs.astral.sh/ruff/rules/useless-if-else/) covers this for the second example (if expressions) but not the first example (if statements)

**Comment by jack-mcivor** [comment]
This is somewhat related to https://github.com/astral-sh/ruff/issues/13929 (Detect empty conditional statements)

**Comment by jack-mcivor** [comment]
> detect if the code on both sides of an if statement are identical

Ideally it could also detect duplicated code blocks across all branches (looking into elifs also). An auto-fix to merge these would be good.

```python
if condition1:
    x = 1
elif condition2:
    x = 2
else:
    x = 2
```
Becomes
```python
if condition1:
    x = 1
else:
    x = 2
```

It seems like this could be included in the existing rule [useless-if-else (RUF034)](https://docs.astral.sh/ruff/rules/useless-if-else/#useless-if-else-ruf034)
> Checks for useless if-else conditions with identical arms.
